From c2e858cd5066aba9594e011578371e7cbb618e16 Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Mon, 11 Jun 2012 22:49:31 +0200
Subject: [PATCH 7/8] pvops: respect 'removable' xenstore flag for block
 devices

Especially this is needed by pmount to allow mount qvm-block attached devices
by normal user.
---
 drivers/block/xen-blkfront.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/block/xen-blkfront.c b/drivers/block/xen-blkfront.c
index 5babe575c288..396875708ccf 100644
--- a/drivers/block/xen-blkfront.c
+++ b/drivers/block/xen-blkfront.c
@@ -2318,6 +2318,7 @@ static void blkfront_connect(struct blkfront_info *info)
 	unsigned long long sectors;
 	int err, i;
 	struct blkfront_ring_info *rinfo;
+	int removable;
 
 	switch (info->connected) {
 	case BLKIF_STATE_CONNECTED:
@@ -2382,6 +2383,12 @@ static void blkfront_connect(struct blkfront_info *info)
 		}
 	}
 
+	err = xenbus_gather(XBT_NIL, info->xbdev->otherend,
+			    "removable", "%d", &removable,
+			    NULL);
+	if (!err && removable)
+		info->vdisk_info |= VDISK_REMOVABLE;
+
 	err = xlvbd_alloc_gendisk(sectors, info);
 	if (err) {
 		xenbus_dev_fatal(info->xbdev, err, "xlvbd_add at %s",
-- 
2.50.1

